views/nexthome.tmpl

```diff
@@ -2,253 +2,147 @@
+                        <div class="heading">PoW Reward</div>
+                        <div class="data mono lh1rem fs14-decimal count fs24 p03rem0">
+                            <span id="bsubsidy-pow">{{template "decimalParts" (float64AsDecimalParts (toFloat64Amount .NBlockSubsidy.PoW) 8 true 2)}}</span>
+                            <span class="pl-1 unit lh15rem">DCR</span>
                         </div>
                     </div>
-                    {{end}}
-  		        </div>
-                <a class="no-underline" href="/blocks"><div class="showmore"><p>View BlockChain History</p></div></a>
+                    <div class="col-sm-4">
+                        <div class="heading">Hashrate</div>
+                        <div class="data mono lh1rem fs14-decimal count fs24 p03rem0">
+                            <span id="hashrate">{{template "decimalParts" (float64AsDecimalParts .HashRate 8 true 2)}}<span>
+                            <span class="pl-1 unit lh15rem">Ph/s</span>
+                        </div>
+                        <div class="subdata">
+                            <span id="hashrate-subdata" class="count">({{template "fmtPercentage" .HashRateChange}} in 24h)</span>
+                        </div>
+                    </div>
+                </div>
             </div>
+            {{end}}
+        </div>
+        <div class="flex-space"></div>
+        <div class="blocks-section">
+            <a class="no-underline title font-weight-bold" href="/blocks">Decred Blockchain</a>
+            <div class="blocks-holder">
+            </div>
+            <a class="no-underline" style="display: block; margin-top: 25px;" href="/blocks"><div class="showmore"><p>View BlockChain History</p></div></a>
         </div>
     </div>
 
+    <script src="/js/tippy.all.min.js"></script>
+    <script type="text/javascript" src="/js/controllers/blocks.js" data-turbolinks-eval="false"></script>
+    <script type="text/javascript" src="/js/controllers/nexthome-socket-update.js" data-turbolinks-eval="false"></script>
     <script>
-        $(document).ready(function(){
-            var count = 0;
-            $(".decBlockchainWrapper .decredblockWrap").click(function(){
-                $(this).toggleClass("expanded");
-                count++;
-
-                if (count > 1) {
-                    $(".decBlockchainWrapper .decredblockWrap").removeClass("expanded");
-                    count = 0;
-                }
-            });
+        // TODO: confirm if .Info.NBlockSubsidy actually represents mempool rewards
+        mempoolSubsidy = {{.Info.NBlockSubsidy}}
 
-            $(".decredblockWrap").click(function() {
-                if ($( ".decredblockWrap" ).hasClass( "expanded" )) {
-                    $( ".decredblockWrap" ).siblings().addClass("fade");
-                } else {
-                    $( ".decredblockWrap" ).siblings().removeClass("fade");
-                }
-            });
-        })
+        blocks = {{.Blocks}};
```

Comment:
Please fetch JSON data via the api. 
Inlining data into the DOM is inefficient and we also don't want to create globals for data that is related to a specific view. 
By fetching the data async, the page can load quickly and show a nice loading animation instead of instead of freezing up.

Reply:
I discussed this with @raedah. I'll move all data fetching operations to API calls for the /nexthome page. I recommended the same be done for other pages but in a separate PR

Reply:
For larger data objects, that makes sense, but the site needs to work without JS.  There's not much point in doing an API request for a handful of values when there available at template execution time.  An array of blocks data, yes that makes sense.

Reply:
Think @chappjc chappjc has a good point here about minimizing implementation complexity. I'm ok with either, follow your gut!